superscore/tests/test_backend.py

```diff
@@ -202,3 +202,16 @@ def test_update_entry(backends: _Backend):
     p1 = Parameter()
     with pytest.raises(BackendError):
         backends.update_entry(p1)
+
+
+@pytest.mark.parametrize("filestore_backend", [("linac_data",)], indirect=True)
+def test_gather_reachable(filestore_backend: _Backend):
+    # top-level snapshot
+    reachable = filestore_backend._gather_reachable(UUID("06282731-33ea-4270-ba14-098872e627dc"))
+    assert len(reachable) == 32
+    assert UUID("927ef6cb-e45f-4175-aa5f-6c6eec1f3ae4") in reachable
+
+    # direct parent snapshot
+    reachable = filestore_backend._gather_reachable(UUID("2f709b4b-79da-4a8b-8693-eed2c389cb3a"))
+    assert len(reachable) == 3
+    assert UUID("927ef6cb-e45f-4175-aa5f-6c6eec1f3ae4") in reachable
```

Comment:
I went about verifying that the `functools.cache` scope behaved as you claimed by augmenting this test case a bit.  Things worked as advertised, and `_gather_reachable` seems to work well when handed an `Entry` as well.
 
```suggestion
    assert UUID("927ef6cb-e45f-4175-aa5f-6c6eec1f3ae4") in reachable
    
    entry = filestore_backend.get_entry(UUID("2f709b4b-79da-4a8b-8693-eed2c389cb3a"))
    assert len(filestore_backend._gather_reachable(entry)) == 3
    entry.children = []
    assert len(filestore_backend._gather_reachable(entry)) == 1
    filestore_backend.update_entry(entry)
    reachable = filestore_backend._gather_reachable(UUID("2f709b4b-79da-4a8b-8693-eed2c389cb3a"))
    assert len(reachable) == 1
```

Reply:
I don't think a cache is ever actually created in this test.  `_gather_reachable` isn't cached on its own, `FilestoreBackend.search` caches it within its method body, so we need `.search` calls to test caching behaviour.  I've moved this suggestion into `test_client::test_search_entries_by_ancestor`.

Reply:
You are correct 👍